src/ds3.c

```diff
@@ -742,9 +758,14 @@ ds3_error* ds3_create_client_from_env(ds3_client** client) {
 
 static void _set_map_value(GHashTable* map, const char* key, const char* value) {
     gpointer escaped_key = (gpointer) _escape_url(key);
-    gpointer escaped_value = (gpointer) _escape_url(value);
-
+    gpointer escaped_value;
+    if(value != NULL) {
+        escaped_value = (gpointer) _escape_url(value);
+    } else {
+        escaped_value = (gpointer) value;
```

Comment:
If value is `NULL` why not just set `escaped_value` to `NULL` instead of casting the already `NULL` pointer?